packages/discovery-provider/plugins/pedalboard/apps/backfill-audio-analyses/src/backfill.ts

```diff
@@ -0,0 +1,223 @@
+      };
+      const signature = generateSignature(dataToSign, config.delegatePrivateKey);
+      const queryParams = new URLSearchParams({ signature });
+      const response = await axios.post<AnalysisResult>(`${analysisUrl}?${queryParams.toString()}`, null, { timeout: REQUEST_TIMEOUT });
+      if (response.status === 200) {
+        const analysisResult = response.data;
+        let audioAnalysisStatus = analysisResult.audio_analysis_status || '';
+
+        // Handle polling if analysis is in progress
+        if (audioAnalysisStatus === '' && analysisResult.status === 'audio_analysis') {
+          const timeout = Date.now() + 90 * 1000; // 1.5 minutes timeout
+          while (Date.now() < timeout) {
+            await new Promise(resolve => setTimeout(resolve, POLL_STATUS_INTERVAL));
+            const pollResponse = await axios.get<AnalysisResult>(`${contentNodeEndpoint}/uploads/${audioUploadId}`, { timeout: REQUEST_TIMEOUT });
+            if (pollResponse.status === 200) {
+              const pollResult = pollResponse.data;
+              if (pollResult.audio_analysis_status === 'done') {
+                audioAnalysisStatus = 'done';
+                analysisResult.audio_analysis_results = pollResult.audio_analysis_results;
+                break;
+              }
+            }
+          }
+        }
+
+        // Handle the result based on analysis status
+        if (audioAnalysisStatus === 'timeout' || audioAnalysisStatus === 'error' || (audioAnalysisStatus === '' && analysisResult.status === 'audio_analysis')) {
+          // Audio analysis job ran and errored, or we timed out waiting for an update from the content node
+          await writeToChain(libs, track.track_id, DEFAULT_MUSICAL_KEY, DEFAULT_BPM, (track.audio_analysis_error_count || 0) + 1);
+        } else if (audioAnalysisStatus === 'done') {
+          const bpm = analysisResult.audio_analysis_results.bpm || DEFAULT_BPM;
+          const key = analysisResult.audio_analysis_results.key || DEFAULT_MUSICAL_KEY;
+          await writeToChain(libs, track.track_id, key, bpm, track.audio_analysis_error_count || 0);
+        }
+      } else {
+        console.log(`Received ${response.status} response from ${contentNodeEndpoint}. Trying next node in rendezvous order`)
+        continue
+      }
+    } catch (error: any) {
+      if (error.isAxiosError !== undefined && error.code === 'ECONNABORTED') {
+        console.log(`Timeout error analyzing audio for track ID ${track.track_id}: ${error.message}`);
+        continue
+      } else {
+        console.error(`Error analyzing audio for track ID ${track.track_id}, upload ID ${track.audio_upload_id}: ${(error as Error).message}`);
+        break
+      }
+    }
+
+    release(); // Release the semaphore permit
+  }
+}
+
+async function fetchTracks(offset: number, limit: number, nodeId: number, db: Knex): Promise<Track[]> {
+  return await db<Track>('tracks')
+    .whereNull('musical_key')
+    .whereNull('bpm')
```

Comment:
Is it possible that only either musical_key or bpm is null?

Reply:
good call, added in https://github.com/AudiusProject/audius-protocol/pull/8788/commits/c982665320ec617c6df43448fc879ae8c793723b

but actually this plugin is only concerned with triggering analyses that haven't been triggered yet. If only one value is present that means an audio analysis error occurred so `audio_analysis_error_count` would be nonzero. I'm leaving that for the discovery repair task to handle (see updated PR description)